### PR TITLE
build: Run executable tests with WasmKit

### PR DESCRIPTION
**Pull request by kateinoigakukun**
This PR adds a new product, WasmKit, which is a Wasm runtime used for running executable tests. And also configures test suites to be run on CI.

Note that the product is not shipped as a part of the toolchain, but is used only for testing purposes.

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
preset=buildbot_linux_crosscompile_wasm
@swift-ci Please test with preset Linux Platform

**Comment by kateinoigakukun** [comment]
preset=buildbot_linux_crosscompile_wasm
@swift-ci Please test with preset Linux Platform

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

**Review by MaxDesiatov — APPROVED** [review]
This is great, thank you so much!

File: utils/swift_build_support/swift_build_support/products/wasmkit.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# swift_build_support/products/wasmkit.py ------------------------------------
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
```
Comment (shahmishal) [inline utils/swift_build_support/swift_build_support/products/wasmkit.py:5]: Can we update the year?

Comment (kateinoigakukun) [reply utils/swift_build_support/swift_build_support/products/wasmkit.py]: Oops, updated.

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

**Pull request merged**